File: src/Climbing Stairs.cpp
#include <vector>
#include <iostream>
#include <stack>
#include <string>
#include <sstream>
using namespace std;

class Solution {
public:
	int climbStairs(int n) {
		vector<int> ret(n+1, 0);
		ret[0] = ret[1] = 1;
		for(int i=2; i<=n; i++)
		{
			ret[i] = ret[i-2] + ret[i-1];
		}
		return ret[n];
	}
};

int main()
{
	Solution so;
	
	return 0;
}

File: python/Climbing Stairs.py
class Solution:
    # @param n, an integer
    # @return an integer
    def climbStairs(self, n):
        ret = [0] * (n+1)
        ret[0] = ret[1] = 1
        for i in range(2, n+1):
            ret[i] = ret[i-1] + ret[i-2]
        return ret[n]